leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -334,22 +331,6 @@ boolean getAndRecordLeaderPingResult(@Nullable Future<Boolean> pingFuture) throw
                     log.warn("unable to get uuid from server", e);
                 }
             }
-
-            // poll for extra completed futures
-            Future<Entry<String, PingableLeader>> future;
-            while ((future = pingService.poll()) != null) {
```

Comment:
In theory, if the ping was interrupted you can still cache the results of any computations that finished in between the interruption and its handler finishing, which looks like what this does - so strictly speaking this does have a point, though I agree it's not worth the complexity.